File: fibonacci.py
import time

def fib(n):
    a = 0
    b = 1
    for _ in range(1, n):
        c = a + b
        a = b
        b = c

with open("fib_results_python.csv", "w") as outfile:
    outfile.write("steps,time_ms\n")

    for i in range(1000, 1000000, 1000):
        start = time.perf_counter()

        fib(i)
        end = time.perf_counter()
        ms = (end - start) * 1000
        outfile.write(f"{i},{ms:.6f}\n") 
        outfile.flush()

File: fibonacci.cpp
#include <iostream>
#include <gmpxx.h>
#include <fstream>
#include <chrono> 


void fib(int n) {
    mpz_class a = 0;
    mpz_class b = 1;
    mpz_class c;
    
    for (int i = 1; i < n; ++i) {
        c = a + b;
        a = b;
        b = c;
    }
}
int main() {
    std::ofstream outfile("fib_results.csv");
    outfile << std::fixed << std::setprecision(6);
    outfile << "steps,time_ms\n";

    for(int i = 1000; i < 1000000; i += 1000) {
        auto start = std::chrono::high_resolution_clock::now();

        fib(i);

        auto end = std::chrono::high_resolution_clock::now();
        double ms = std::chrono::duration<double, std::milli>(end - start).count();
        outfile << i << "," << ms << "\n";
    }
    return 0;
}